lib/buffer.js

```diff
@@ -503,12 +503,12 @@ Buffer.prototype.copy = function(target, targetStart, sourceStart, sourceEnd) {
   return binding.copy(this, target, targetStart, sourceStart, sourceEnd);
 };
 
-Buffer.prototype.toString = function() {
+Buffer.prototype.toString = function(encoding, start, end) {
   let result;
   if (arguments.length === 0) {
     result = this.utf8Slice(0, this.length);
   } else {
-    result = slowToString.apply(this, arguments);
+    result = slowToString.call(this, encoding, start, end);
```

Comment:
This is the only call site for `slowToString`, maybe we can drop the `.call` entirely?